src/kafka-net/ProtocolGateway.cs

```diff
@@ -0,0 +1,139 @@
+﻿using KafkaNet.Model;
+using KafkaNet.Protocol;
+using System;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace KafkaNet
+{
+    public class ProtocolGateway : IDisposable
+    {
+        private readonly IBrokerRouter _brokerRouter;
+        //Add Loger
+        public ProtocolGateway(params Uri[] brokerUrl)
+        {
+            var kafkaOptions = new KafkaOptions(brokerUrl) { MaximumReconnectionTimeout = TimeSpan.FromSeconds(60), ResponseTimeoutMs = TimeSpan.FromSeconds(60) };
+            _brokerRouter = new BrokerRouter(kafkaOptions);
+        }
+
+        public ProtocolGateway(IBrokerRouter brokerRouter)
+        {
+            _brokerRouter = brokerRouter;
+        }
+        public ProtocolGateway(KafkaOptions kafkaOptions)
+        {
+            _brokerRouter = new BrokerRouter(kafkaOptions);
+        }
+
+        private readonly int _maxRetry = 2;
+
+        /// <exception cref="InvalidTopicMetadataException">Thrown if the returned metadata for the given topic is invalid or missing.</exception>
+        /// <exception cref="InvalidPartitionException">Thrown if the give partitionId does not exist for the given topic.</exception>
+        /// <exception cref="ServerUnreachableException">Thrown if none of the Default Brokers can be contacted.</exception>
+        /// <exception cref="ResponseTimeoutException">Thrown if none of the Default Brokers can be contacted.</exception>
+        ///     /// <exception cref="SocketException">Thrown if none of the Default Brokers can be contacted.</exception>
+        /// <exception cref="KafkaApplicationException">Thrown if none of the Default Brokers can be contacted.</exception>
+        public async Task<T> SendProtocolRequest<T>(IKafkaRequest<T> request, string topic, int partition) where T : class,IBaseResponse
+        {
+            //Log Start  Sesion  Id
```

Comment:
there were more, please go over that file and remove them (or replace with TODO)